src/main/java/com/workshop/demo/model/Restaurant.java

```diff
@@ -0,0 +1,169 @@
+package com.workshop.demo.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.LongSummaryStatistics;
+import java.util.stream.Collectors;
+
+import org.hibernate.annotations.NaturalId;
+
+import com.fasterxml.jackson.annotation.JsonIdentityInfo;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+import com.workshop.demo.model.audit.UserDateAudit;
+
+import java.util.Collections;
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+import jakarta.persistence.UniqueConstraint;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@EqualsAndHashCode(callSuper = true)
+@Entity
+@Data
+@Table(name = "Restaurants", uniqueConstraints = { @UniqueConstraint(columnNames = { "name" }),
+        @UniqueConstraint(columnNames = { "email" }) })
+@JsonIgnoreProperties(value = { "updatedBy" })
+@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
+public class Restaurant extends UserDateAudit {
+    private static final long serialVersionUID = 1L;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "name")
+    private String name;
+
+    @NotBlank
+    @Column(name = "score")
```

Comment:
Do we plan to update the score in restaurant periodically? Otherwise since we are interested in the average of the score, we can always get it by aggregating from Review.